### ISSUE TITLE
Add examples of .NET CLI install for travisci/appveyor

### ISSUE THREAD

**Issue by enricosada** [dotnet/cli#2113]
Lots of OSS libraries use [travis-ci](https://travis-ci.org/) and/or [appveyor](https://www.appveyor.com) as ci server (usually integrated with github)

The install doc is really nice, but it's possibile to have a working example (under ci) ?
Last week a version change broke the build script (already fixed), but it's nice to have a working implementation to copy :smile: 

I use:

- https://github.com/enricosada/fsharp-dotnet-cli-samples/blob/master/.travis.yml
- https://github.com/enricosada/fsharp-dotnet-cli-samples/blob/master/appveyor.yml

this install .NET CLI (latest and/or pinned version) and add to path

it's pretty much download `scripts/obtain/install.ps1` ( or .sh ) and execute

if ok about adding appveyor/travis i can send a pr, but this repo is not the right place i think, i found https://github.com/dotnet/core but maybe it's https://github.com/dotnet/core-docs ? :confused:

**Comment by piotrpMSFT** [comment]
@blackdwarf can you provide some guidance? These would be great samples to have somewhere!

**Comment by blackdwarf** [comment]
@piotrpMSFT sure, put it on my backlog, seems slightly empty-ish at the moment. 

All jokes aside, @enricosada: TravisCI allows you to use apt-get AFAIK. Are people in general using this method? One thing that I'm wondering with the script install is getting dependencies installed. Did you run into this. 

And yes, in general, core-docs is the repo where our docs live.

**Comment by akoeplinger** [comment]
Travis allows apt-get but only for the images where sudo is available (or if the apt feed is whitelisted).

I'm one of the maintainers of the C# support for Travis (i.e. if you specify `language: csharp` in the .travis.yml) which is currently using Mono. We don't have that restriction in the language setup script, so I'm planning on using the apt feed to support .NET Core once RC2 is ready :smile:

**Comment by enricosada** [comment]
it's a pleasure to give you more work @blackdwarf :smile: 

I tried installers/packages time ago, but got issues (probably pebkac), and was more difficult to test locally
I like binaries because it's easier to reproduce locally (it's just a download and unzip)
It's also possibile to use the installer in appveyor, if we want (or it's better) to use packages.

But it's easier to show the code :smile: 
I'll open a pr in `core-docs` repo this weekend with appveyor and travis, and we can add to pr  review the travis and appveyor team for insight

i'll build a project in [samples directory](https://github.com/dotnet/core-docs/blob/master/samples) or a demo run (`dotnet new` => `dotnet restore` => `dotnet build` => `dotnet pack`)

travis/appveyor are really nice for oss projects, and a clean example help getting started with .NET CLI.

**Comment by enricosada** [comment]
i was thinking about you @akoeplinger for review, that was fast

**Comment by blackdwarf** [comment]
@enricosada thanks for the PR, that would be super helpful. Let's take it there then. I will be making a large-ish CLI docs PR this week, so if you want, you can wait for that to settle or you can contribute to the "samples" directory in there and I can take care of the text around it (as in, explanations, scenarios and what not). 

Sounds like a plan?

**Comment by blackdwarf** [comment]
@akoeplinger yeah, thought as much. Again, the net effect is  the same apart from the dependencies. Speaking of which, lemme go back to documenting those for this repo...

**Comment by jskeet** [comment]
@akoeplinger: Could you ping this bug (possibly with a link to more information) when a non-sudo-requiring environment is available? It would be much appreciated. (Admittedly a `language: dotnet` with dotnet already installed would be even better...)

**Comment by akoeplinger** [comment]
@jskeet sure. I'm currently quite busy with a few last minute Mono 4.4 bug fixes as we're releasing soon, but I'll try to fit it in next week.

FWIW, there was a lengthy discussion when we added C# to travis about `language: dotnet` vs `language: csharp (or fsharp/vb)` but we decided in the end that .NET isn't a language so we picked the other option ;)

Adding out of the box support for .NET Core to Travis is tracked here: https://github.com/travis-ci/travis-ci/issues/5189

**Comment by enricosada** [comment]
sry i forgot about this issue :smile: 

so as we discussed i added in core-docs repo the example for both travis and appveyor.
That's what i use. 
I prefer sdk binaries instead of sdk installer because are easier to reproduce locally (no fear, it's just an unzip) and no permission needed, so can be copied everywhere.
I didnt use the dotnet/cli install script because it's more work than just curl | unzip to understand, and i had previously issues when was changed (ok we can pin to a specific commit)

**Comment by tmds** [comment]
Hi all, I setup Travis to build my dotnet project on linux. You can find my configuration at [.travis.yml](https://github.com/tmds/Tmds.DBus/blob/master/.travis.yml). I started from the travis.yml file I found here. One thing I changed was setting the language to `generic`. This is an undocumented feature.

I am looking for some info on how to get the build artifacts on a NuGet feed.

**Comment by blackdwarf** [comment]
I believe that https://docs.microsoft.com/en-us/dotnet/articles/core/tools/using-ci-with-cli contains the needed samples. If not, the issue should be filed against dotnet/core-docs, as this is a documentation issue.

**Comment by AnderssonPeter** [comment]
the page you are referring to only says **TODO** in the TravisCI section.

**Comment by dustinmoris** [comment]
Hey guys I wanted to inform you that TravisCI has a new trusty beta environment and supports .NET Core out of the box now. I have written an [elaborate blog post about it](https://dusted.codes/building-and-shipping-a-dotnet-core-application-with-docker-and-travisci) and also included an example how to deploy the built application as a Docker image to a Docker registry like the Docker hub. Hope that helps.

**Comment by enricosada** [comment]
@dustinmoris the disable telemetry option is nice, it speedup things?.

Maybe is worth add that to the travisci example in docs?
I added an example of travis in the docs, see https://docs.microsoft.com/en-us/dotnet/articles/core/tools/using-ci-with-cli

The docs contains a link with an example .travis.yml (with an osx/ubuntu matrix too)
